testsuite/smoke-test/src/full_nodes.rs

```diff
@@ -1,292 +1,285 @@
-        get_balances(&mut vfn_client, 0)
-    ));
-
-    let sequence = get_and_reset_sequence_number(&mut vfn_client, sender_account);
-    wait_for_transaction(&mut validator_client, sender_account, sequence - 1);
-    assert!(compare_balances(
-        vec![(10.0, XUS.to_string())],
-        get_balances(&mut validator_client, 0),
-    ));
-
-    // reset sequence number for sender account
-    get_and_reset_sequence_number(&mut validator_client, sender_account);
-    get_and_reset_sequence_number(&mut vfn_client, sender_account);
-    get_and_reset_sequence_number(&mut validator_client, creation_account);
-    get_and_reset_sequence_number(&mut vfn_client, creation_account);
-
-    // mint from validator and check both nodes have correct balance
-    validator_client.create_next_account(false).unwrap();
-    vfn_client.create_next_account(false).unwrap();
-    pfn_client.create_next_account(false).unwrap();
-
-    mint_coins(&mut validator_client, 1, 10, XUS, true);
-    let sequence = get_and_reset_sequence_number(&mut validator_client, sender_account);
-    wait_for_transaction(&mut vfn_client, sender_account, sequence - 1);
-
-    assert!(compare_balances(
-        vec![(10.0, XUS.to_string())],
-        get_balances(&mut validator_client, 1)
-    ));
-    assert!(compare_balances(
-        vec![(10.0, XUS.to_string())],
-        get_balances(&mut vfn_client, 1)
-    ));
-
-    // minting again on validator doesn't cause error since client sequence has been updated
-    mint_coins(&mut validator_client, 1, 10, XUS, true);
-
-    // test transferring balance from 0 to 1 through full node proxy
-    transfer_coins(&mut vfn_client, 0, 1, 10, XUS, true);
-
-    assert!(compare_balances(
-        vec![(0.0, XUS.to_string())],
-        get_balances(&mut vfn_client, 0)
-    ));
-    assert!(compare_balances(
-        vec![(30.0, XUS.to_string())],
-        get_balances(&mut validator_client, 1)
-    ));
-
-    let sequence = get_and_reset_sequence_number(&mut validator_client, account);
-    wait_for_transaction(&mut pfn_client, account, sequence - 1);
-    assert!(compare_balances(
-        vec![(0.0, XUS.to_string())],
-        get_balances(&mut pfn_client, 0)
-    ));
-    assert!(compare_balances(
-        vec![(30.0, XUS.to_string())],
-        get_balances(&mut pfn_client, 1)
-    ));
+    vfn_client
+        .wait_for_signed_transaction(&txn, None, None)
+        .unwrap();
+    assert_balance(&vfn_client, &account_0, 7);
+    assert_balance(&vfn_client, &account_1, 13);
+
+    pfn_client
+        .wait_for_signed_transaction(&txn, None, None)
+        .unwrap();
+    assert_balance(&pfn_client, &account_0, 7);
+    assert_balance(&pfn_client, &account_1, 13);
 }
 
 #[test]
 fn test_vfn_failover() {
-    let mut env = SmokeTestEnvironment::new(7);
-    env.setup_vfn_swarm();
-    let vfn_swarm = env.vfn_swarm();
-    env.add_public_fn_swarm(PUBLIC, 1, None, &vfn_swarm.lock().config);
+    let mut swarm = new_local_swarm(4);
```

Comment:
So, this was originally 7, because of the problem that each node needed to connect to other nodes.  And with a small enough pool, some VFNs actually get starved of backup connections to other nodes.

Every VFN connects to 3 other VFNs, and prevents > 3 VFNs from connecting to them.

Reply:
So if we had 7, then lots of the comments in the test were wrong since they were stating that they were killing nodes >f in order to stop consensus. With 7 consensus wasn't being stopped.